test/core/test_dlfcn_jspi.c

```diff
@@ -0,0 +1,29 @@
+// Copyright 2025 The Emscripten Authors.  All rights reserved.
+// Emscripten is available under two separate licenses, the MIT license and the
+// University of Illinois/NCSA Open Source License.  Both these licenses can be
+// found in the LICENSE file.
+
+#include <emscripten.h>
+#include <stdio.h>
+#include <dlfcn.h>
+
+EM_ASYNC_JS(int, test, (), {
+  console.log("sleeping");
+  await new Promise(res => setTimeout(res, 500));
```

Comment:
I'd say either make the timeout 0 or 1. Either will give us a turn of the event loop and should test what we want. No reason the artificially slow down the test.

Reply:
I originally set it this high during manual testing since I wanted a perceptible delay. And then I forgot about it.